react-backend/client/src/components/visitors/qrcode.js

```diff
@@ -44,13 +44,9 @@ function instascan() {
         if (cameras.length > 0) {
           scanner.start(cameras[0]);
         } else {
-          throw new Error('ERROR HAPPENING AT getCameras');
+          reject('ERROR HAPPENING AT getCameras');
         }
       })
-      .catch(err => {
```

Comment:
You still need to have a `.catch` here, whenever a function returns a `Promise` it needs to have a `.then` and a `.catch`. Inside of the `.catch` here you could `reject` since this is all happening inside of a `Promise` you are writing. Then when you call your `instascan()` function, any failures inside of it will trigger the `.catch` you have on [line 111](https://github.com/FAC-11/DataPower/blob/b3eceabfdb304e1633a3711c884a93056be5aa26/react-backend/client/src/components/visitors/qrcode.js#L111)